lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -49,54 +49,76 @@ BOOL MayHaveSideEffectOnNode(ParseNode *pnode, ParseNode *pnodeSE)
         return false;
     }
 
-    uint fnop = ParseNode::Grfnop(pnodeSE->nop);
-    if (fnop & fnopLeaf)
-    {
-        // pnodeSE is a leaf and can't kill anything.
-        return false;
-    }
+    ArenaAllocator *alloc = byteCodeGenerator->GetAllocator();
+    SList<ParseNode*> pNodeSEStack(alloc);
```

Comment:
>SList<ParseNode*> pNodeSEStack(alloc); [](start = 4, length = 38)

Using a stack for this does a DFS, could we use a queue for a BFS instead? Is there any heuristic about which way might terminate sooner? Looks like we have early termination conditions (return true) so we might be able to choose which way will terminate sooner.